group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/TargetAssignmentBuilder.java

```diff
@@ -252,14 +252,16 @@ public TargetAssignmentResult build() throws PartitionAssignorException {
             if (updatedMemberOrNull == null) {
                 memberSpecs.remove(memberId);
             } else {
-                ConsumerGroupMember member = members.get(memberId);
-                Assignment assignment;
+                Assignment assignment = targetAssignment.getOrDefault(memberId, Assignment.EMPTY);
+
                 // A new static member joins and needs to replace an existing departed one.
-                if (member == null && staticMembers.containsKey(updatedMemberOrNull.instanceId())) {
-                    assignment = targetAssignment.getOrDefault(staticMembers.get(updatedMemberOrNull.instanceId()), Assignment.EMPTY);
-                } else {
-                    assignment = targetAssignment.getOrDefault(memberId, Assignment.EMPTY);
+                if (updatedMemberOrNull.instanceId() != null) {
+                    String previousMemberId = staticMembers.get(updatedMemberOrNull.instanceId());
+                    if (previousMemberId != null) {
+                        assignment = targetAssignment.getOrDefault(updatedMemberOrNull.instanceId(), Assignment.EMPTY);
```

Comment:
not sure whether `updatedMemberOrNull.instanceId()` should be replaced by `previousMemberId`?

Reply:
Good catch! The unit actually worked because it uses the same id for both the member id and the static id. I have updated updated it.